worlds/banjo_tooie/Hints.py

```diff
@@ -0,0 +1,578 @@
+            bad_location_names.append(sorted_silos.pop())
+
+    if world.options.cheato_rewards:
+        bad_location_names.extend([
+            locationName.CHEATOR3,
+        ])
+
+    if world.options.honeyb_rewards:
+        bad_location_names.extend([
+            locationName.HONEYBR4,
+        ])
+    return bad_location_names
+
+
+def generate_move_hints(world: "BanjoTooieWorld", hints: List[Hint]):
+    move_locations = get_move_locations(world)
+    for location in move_locations:
+        hints.append(Hint(world, location))
+
+
+def get_move_locations(world: "BanjoTooieWorld") -> List[Location]:
+    all_moves_names = []
+
+    # We don't want BT moves to be hinted when they're in the vanilla location.
+    if world.options.randomize_bt_moves:
+        all_moves_names.extend(moves_table.keys())
+    all_moves_names.extend(bk_moves_table.keys())
+    all_moves_names.extend(progressive_ability_table.keys())
+
+    all_move_locations = [location for location in get_all_hintable_locations(world)
+                          if location.item.name in all_moves_names and location.item.player == world.player]
+    world.random.shuffle(all_move_locations)
+    selected_move_locations = []
+
+    for location in all_move_locations:
+        if len(selected_move_locations) >= min(
+            world.options.signpost_move_hints.value,
+            world.options.signpost_hints.value
+        ):
+            return selected_move_locations
+        selected_move_locations.append(location)
+    return selected_move_locations
+
+
+def get_location_by_name(world: "BanjoTooieWorld", name: str) -> Location | None:
+    potential_match = list(filter(lambda location: location.name == name, get_player_hintable_locations(world)))
+    if potential_match:
+        return potential_match[0]
+    return None
+
+
+def get_all_hintable_locations(world: "BanjoTooieWorld") -> List[Location]:
+    return [location for location in world.multiworld.get_locations() if should_consider_location(location)]
+
+
+def get_player_hintable_locations(world: "BanjoTooieWorld") -> List[Location]:
+    return [location for location in world.multiworld.get_locations(world.player) if should_consider_location(location)]
+
+
+def should_consider_location(location: Location) -> bool:
+    if not location.item or not location.address:
+        return False
+    if location.progress_type == LocationProgressType.EXCLUDED:
+        return False
+
+    location_hint_blacklist = [
+        *WorldUnlocks_table.keys(),
+        *MumboTokenBoss_table.keys(),
+        *MumboTokenGames_table.keys(),
+        *MumboTokenJinjo_table.keys(),
+    ]
+    if location.name in location_hint_blacklist:
+        return False
+    return True
+
+
+def get_signpost_location_ids() -> List[int]:
+    location_datas = list(filter(lambda location_data: location_data.group == "Signpost", all_location_table.values()))
+    return [location_data.btid for location_data in location_datas]
```

Comment:
```suggestion
    return [location_data.btid for location_data in all_location_table.values() if location_data.group == "Signpost"]
```
This one can also be condensed into a single expression.